src/basic/terminal-util.c

```diff
@@ -891,33 +891,24 @@ bool on_tty(void) {
 }
 
 int getttyname_malloc(int fd, char **ret) {
-        size_t l = 100;
+        char path[PATH_MAX], *c; /* PATH_MAX is counted *with* the trailing NUL byte */
```

Comment:
i must say I never really liked the approach of allocating this much stuff on the stack (it's a whole page after all, if we lived in kernel space that'd be more than half of what's actually available at all!) without actually needing it. 

The nice thing about the VLA approach was that we only allocated as much as we needed (well, rounded up to the next 2 exponent of 100), which I liked. 

Now the VLA thing sucks admittedly, but I must say the fixed stack thing isn#t too pretty either... We usually avoided PATH_MAX sized buffers in most of our code so far, and instead allocated with the correct size.

Anyway, I figure to a point this is a matter of taste I guess, dunno. Either way, it made me propose #12076 at least, which does without this.

Reply:
We should be going the opposite direction and ensuring all paths be *safely* dynamically sized.

Systems aren't getting smaller, the hardware density only increases and systems accordingly get larger.  VMs and now containers are results of this, and with containers in particular it's obviously very likely for the host to be traversing paths composed of at least a host *and* container worth of filesystem in a single call.  If we go down the nested containers route, the paths can plausibly become a multiple of PATH_MAX starting from a root at the host.

I think we should be viewing PATH_MAX as just another obsolete vestige of UNIX's simpler days like select() and its FD_SETSIZE.

Reply:
In general, probably yes. But in this particular case, ttypath only makes sense within the scope of a single machine, and no container recursion is necessary or possible. And like I said in the review of #12076, we only have two callers which use the returned value in context which cannot deal with arbitrarily long values anyway.